sdk/src/main/java/com/kin/ecosystem/main/view/EcosystemActivity.java

```diff
@@ -247,6 +267,19 @@ public void navigateBack() {
 			super.onBackPressed();
 			overridePendingTransition(0, R.anim.kinecosystem_slide_out_right);
 		} else {
+			BackStackEntry entry = getSupportFragmentManager().getBackStackEntryAt(count - 1);
+			if (entry != null && entry.getName().equals(MARKETPLACE_TO_ORDER_HISTORY)) {
+				// After pressing back from OrderHistory, should put the attrs again.
+				// This is the only fragment that should set presenter again on back.
+				MarketplaceFragment marketplaceFragment = getSavedMarketplaceFragment();
+				if (marketplaceFragment != null) {
```

Comment:
what happens if marketplaceFragment == null?

Reply:
no sure it's even possible, only this fragment goes to back stack.